src/deploy.ts

```diff
@@ -445,6 +456,31 @@ export async function deploy(
   Telemetry.record({event: "deploy", step: "finish"});
 }
 
+async function findOldestSourceAge(effects: DeployEffects, config: Config): Promise<number> {
+  let oldestAge = -Infinity;
+  const nowMs = Date.now();
+
+  try {
+    for await (const file of effects.visitFiles(config.root, {ignoreObservable: false})) {
+      let stat: Stats;
+      const joinedPath = join(config.root, file);
+      try {
+        stat = await effects.stat(joinedPath);
+        if (stat?.isFile()) {
+          if (nowMs - stat.mtimeMs > oldestAge) {
+            oldestAge = nowMs - stat.mtimeMs;
+          }
+        }
+      } catch (error) {
+        // ignore
+      }
```

Comment:
Do we want to ignore these here? Seems like we should just let them throw? (The ceremony around catching errors and re-wrapping them in `CliError` does seem somewhat cumbersome in `findBuildFiles` so I can understand not wanting to repeat that here, but I’m wondering if that’s really necessary to produce a nice error message, or if we could just let things fail a bit more generically if these errors are going to be very rare.)